src/compute-client/src/controller/instance.rs

```diff
@@ -1096,14 +1103,19 @@ where
         // storage metadata needed by the compute instance.
         let mut source_imports = BTreeMap::new();
         for (id, (si, monotonic)) in dataflow.source_imports {
-            let collection = self
+            let collection_metadata = self
+                .storage_controller
+                .collection_metadata(id)
+                .map_err(|_| DataflowCreationError::CollectionMissing(id))?;
+            let collection_description = self
                 .storage_controller
-                .collection(id)
+                .collection_description(id)
                 .map_err(|_| DataflowCreationError::CollectionMissing(id))?;
+
             let desc = SourceInstanceDesc {
-                storage_metadata: collection.collection_metadata.clone(),
+                storage_metadata: collection_metadata.clone(),
                 arguments: si.arguments,
-                typ: collection.description.desc.typ().clone(),
+                typ: collection_description.desc.typ().clone(),
```

Comment:
I don't think they should differ, I just did it like this because we got all the info from storage before. We're still getting the `collection_metadata` from storage, so might as well continue getting both that way?

Reply:
I'd slightly prefer getting rid of these four lines of additional code, and the confusion around if fetching the description again is necessary or not, but I'm also fine with leaving it as is.

Reply:
changing it to your suggestion! This will also allow me to remove the `collection_description()` method ... :sweat_smile: